src/mjolnir/graphenhancer.cc

```diff
@@ -413,7 +413,8 @@ void UpdateTurnLanes(const OSMData& osmdata,
         // Should have a left.
         if (has_turn_left(outgoing_turn_type)) {
           // check for a right.
-          EnhanceRightLane(directededge, tilebuilder, reader, lock, enhanced_tls);
+          if (!directededge.start_restriction())
```

Comment:
Could you add a gurka test for the lane enhancements, showing how it's supposed to work with and without restrictions?

Reply:
Done